### DISCUSSION TITLE
`pageContext` in onBeforeRender() return mostly falsy value.

### DISCUSSION THREAD

**Discussion by BanDroid** [vikejs/vike#969] [Discussion] [answered]
### Description

```jsx
async function onBeforeRender(pageContext) {
  console.log(pageContext.urlParsed);
  const res = await fetch(`${pageContext.urlParsed.origin}/api/posts`);
  const posts = await res.json();
  return {
    pageContext: {
      pageProps: {
        posts
      }
    }
  };
}
```

the console output the `pageContext.urlParsed` like this:

```javascript
{
  origin: null,
  pathname: '/',
  pathnameOriginal: '/',
  search: {},
  searchAll: {},
  searchOriginal: null,
  hash: '',
  hashOriginal: null
}
```

this is rendered in `.page.jsx` not using .server or .client

**Comment by brillout** [comment]
I don't see the problem.

**Reply by BanDroid** [reply]
how to access the pageContext value in onBeforeRender? i need to fetch data and knowing the origin of the url, pageContext.urlParsed.origin is null

**Reply by brillout** [reply, marked as answer]
It's `null` because it's missing in your `renderPage({ urlOriginal })` integation.

**Reply by BanDroid** [reply]
thank you

**Reply by brillout** [reply]
(Btw. for things like that, create a new GitHub Disucssion instead of create a new GitHub Issue. Also check https://github.com/brillout/rules, thanks!)

**Reply by BanDroid** [reply]
ok thanks, sometimes it is ambigous wether it should be an issue or discussion
